### PR TITLE
Add Alluxio AMI files

### PR DESCRIPTION
**Pull request by tieujason330**

### CODE REVIEW

**Comment by tieujason330** [comment]
jenkins, test this please

**Comment by tieujason330** [comment]
@Xenorith PTAL, thanks

File: integration/packer/alluxio-ami/packer.json [hunk 1]
```diff
@@ -0,0 +1,168 @@
+{
+  "variables": {
+    "alluxio_download_url": "",
+    "alluxio_edition_flavor": "",
+    "alluxio_version": "",
```
Comment (Xenorith) [inline integration/packer/alluxio-ami/packer.json:5]: set this to be the current SNAPSHOT version and have it update as part of `dev/scripts/update_versions.sh` so the version gets updated along with all the other versions when release is cut

File: integration/packer/alluxio-ami/packer.json [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        "most_recent": true,
+        "owners": [
+          "137112412989"
+        ]
+      },
+      "ssh_keypair_name": "{{user `ec2_keypair_name`}}",
+      "ssh_private_key_file": "{{user `ec2_keypair_pem_file`}}",
+      "ssh_username": "ec2-user",
+      "tags": {
+        "Origin": "",
```
Comment (Xenorith) [inline integration/packer/alluxio-ami/packer.json:54]: do we need to introduce more variables to optionally populate the tag values? i wonder if it is possible to use a map as a variable value and just say "tags": {{user `tags_map`}}

File: integration/packer/alluxio-ami/src/alluxio.com/factory/command.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	"os/exec"
+	"strings"
+	"time"
+)
+
+func Exec(app string, args []string) (string, error) {
+	log.Println(fmt.Sprintf("%v %v", app, strings.Join(args, " ")))
+	cmd := exec.Command(app, args...)
+	stdout, err := cmd.Output()
+
```
Comment (Xenorith) [inline integration/packer/alluxio-ami/src/alluxio.com/factory/command.go:26]: extra newline

File: integration/packer/alluxio-ami/packer.json (hunk 1 above)
Comment (tieujason330) [reply integration/packer/alluxio-ami/packer.json]: updated

File: integration/packer/alluxio-ami/packer.json (hunk 2 above)
Comment (tieujason330) [reply integration/packer/alluxio-ami/packer.json:50]: packer doesn't allow user variables to be objects

File: integration/packer/alluxio-ami/src/alluxio.com/factory/command.go (hunk 3 above)
Comment (tieujason330) [reply integration/packer/alluxio-ami/src/alluxio.com/factory/command.go]: removed

File: integration/packer/alluxio-ami/packer.json (hunk 2 above)
Comment (Xenorith) [reply integration/packer/alluxio-ami/packer.json:50]: looks like they started introducing the terraform HCL to packer in a newer release: https://www.packer.io/guides/hcl/variables/#maps

guess we can't do this generically yet, but for now we still need to be able to set this `Origin` tag value for cleanup purposes? or do you have another way of setting the tag?

Comment (tieujason330) [reply integration/packer/alluxio-ami/packer.json:50]: we can have the origin tag be a user variable

**Pull request closed without merging**